tools/install.cmake

```diff
@@ -0,0 +1,170 @@
+    if(DEFINED "${library_prefix}_EXTRA_INCLUDE_PUBLIC_DIRS")
+        target_include_directories("${library_name}"
+                        PUBLIC ${${library_prefix}_EXTRA_INCLUDE_PUBLIC_DIRS})
+    endif()
+    if(DEFINED "${library_prefix}_EXTRA_INCLUDE_PRIVATE_DIRS")
+        target_include_directories("${library_name}"
+                        PRIVATE ${${library_prefix}_EXTRA_INCLUDE_PRIVATE_DIRS})
+    endif()
+
+    # Allow a path to a custom config header to be passed through a CMake flag.
+    set(config_prefix "${library_prefix}")
+    if(";${OTA_BACKENDS};" MATCHES ";${library_prefix};")
+        set(config_prefix "OTA")
+    endif()
+    if(DEFINED "${config_prefix}_CUSTOM_CONFIG_DIR")
+        target_include_directories("${library_name}"
+                                    PRIVATE ${${config_prefix}_CUSTOM_CONFIG_DIR})
+    else()
+        target_compile_definitions("${library_name}" PRIVATE -D${config_prefix}_DO_NOT_USE_CUSTOM_CONFIG)
+        # PKCS11 requires a config so include the one from the demos by default.
+        if(${config_prefix} STREQUAL "PKCS")
+            target_include_directories("${library_name}" PRIVATE
+                                        ${DEMOS_DIR}/pkcs11/common/include
+                                        ${LOGGING_INCLUDE_DIRS})
+            target_compile_definitions("${library_name}" PUBLIC
+                                       -DMBEDTLS_CONFIG_FILE="mbedtls_config.h")
+        endif()
+    endif()
+
+    # Add public include directories to library target.
+    if(DEFINED "${library_prefix}_INCLUDE_PUBLIC_DIRS")
+        target_include_directories("${library_name}"
+                                    PUBLIC ${${library_prefix}_INCLUDE_PUBLIC_DIRS})
+        foreach(library_public_dir ${${library_prefix}_INCLUDE_PUBLIC_DIRS})
+            file(GLOB_RECURSE library_headers LIST_DIRECTORIES false ${library_public_dir}/*.h !${library_public_dir}/*private*)
+            list(APPEND ALL_CSDK_PUBLIC_HEADERS ${library_headers})
+        endforeach()
+    endif()
+
+    # Add private include directories to library target.
+    if(DEFINED "${library_prefix}_INCLUDE_PRIVATE_DIRS")
+        target_include_directories("${library_name}"
+                                    PRIVATE ${${library_prefix}_INCLUDE_PRIVATE_DIRS})
+    endif()
+
+    # Install the library target.
+    install(TARGETS "${library_name}" LIBRARY DESTINATION "${CSDK_LIB_INSTALL_PATH}")
+endforeach()
+
+# Install platform abstractions as shared libraries if enabled.
+if(INSTALL_PLATFORM)
+    set(PLATFORM_DIRECTORIES
+            ${COMMON_TRANSPORT_INCLUDE_PUBLIC_DIRS}
+            ${PLATFORM_DIR}/posix/ota_pal/source/include)
+    # Create target for POSIX port of OTA if LIB_RT is installed.
+    if(NOT(${LIB_RT} STREQUAL "LIB_RT-NOTFOUND"))
+        add_library(ota_posix
+                     ${OTA_OS_POSIX_SOURCES})
+        target_link_libraries(ota_posix PUBLIC ${LIB_RT} ota_pal)
```

Comment:
Where is the `ota_pal` library built?

Reply:
In `platform/posix/ota_pal`, and it is an implementation specific to POSIX.